sdk/python/lib/pulumi/dynamic/__main__.py

```diff
@@ -41,6 +45,14 @@
 
 
 def get_provider(props) -> ResourceProvider:
+    # Ensure Settings are configured in the thread that calls get_provider
+    configure(
+        Settings(
+            project=os.environ.get("PULUMI_PROJECT", "project"),
+            stack="stack",
```

Comment:
`stack` and `organization` are the default values already.

Ideally we find away to pass the actual values to the dynamic provider